pkg/apis/serving/v1/revision_validation.go

```diff
@@ -57,7 +58,14 @@ func (r *Revision) Validate(ctx context.Context) *apis.FieldError {
 // Validate implements apis.Validatable
 func (rts *RevisionTemplateSpec) Validate(ctx context.Context) *apis.FieldError {
 	errs := rts.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec")
-	errs = errs.Also(autoscaling.ValidateAnnotations(rts.GetAnnotations()).ViaField("metadata.annotations"))
+	allowZeroInitialScale := false
+	if apisconfig.FromContext(ctx) != nil {
+		autoscalerConfig := apisconfig.FromContext(ctx).Autoscaler
+		if autoscalerConfig != nil {
```

Comment:
Same as above: When would this ever be nil? Same for the config in general.

Reply:
This is for the tests. Would it be better if we make sure all of the test contexts has the config instead of making this change?

Reply:
I think so, yeah. This feels a little weird to reason about in production code.